### PR TITLE
Close OpenRCT2#20831: Display author field on ride overview window

### LINKED ISSUE #20831: Display author field on ride overview window
A good first issue feature requests that people have been asking for. Thisone comes from https://github.com/OpenRCT2/OpenRCT2/discussions/17459.

Currently people are crediting themselves in the titles of rides, which causes a lot of clutter. That's what the author field is for.
The author field is displaying on the object selector, but is not displayed on the ride overview window.

Keep in mind that DAT files do not have an author field, so when selected should also not display this statistic.

### PR DESCRIPTION
**Pull request by fredriktegnell**
Fixes #20831. Added display of author on the right bottom corner of the ride overview window (right above the cost):
image

### CODE REVIEW

**Comment by tupaschoal** [comment]
I wonder whether moving it upwards wouldn't make it less confusing (also adding a "Author" or "Author(s)")
image

Close to the money it looks like it's related to it, somehow

**Comment by fredriktegnell** [comment]
> I wonder whether moving it upwards wouldn't make it less confusing (also adding a "Author" or "Author(s)") image
> 
> Close to the money it looks like it's related to it, somehow

Yeah that sounds like a good idea. Will get to it and then post a update.

**Comment by fredriktegnell** [comment]
Here is an updated version of this, what do you think?
authors5

**Comment by tupaschoal** [comment]
I like it a lot more.

File: src/openrct2-ui/windows/NewRide.cpp [hunk 1]
```diff
@@ -970,6 +970,28 @@ class NewRideWindow final : public Window
...
             DrawTextBasic(dpi, screenPos + ScreenCoordsXY{ textWidth, 51 }, stringId, ft, { TextAlignment::RIGHT });
         }
+
+        // Draw object author (will be blank space if no author in file or a non JSON object)
+        {
+            ft = Formatter();
+            std::string authorsString;
+            auto rideObject = static_cast<RideObject*>(rideEntry->obj);
+            auto repoItem = ObjectRepositoryFindObjectByEntry(&(rideObject->GetObjectEntry()));
+            authorsString.append("Author(s): ");
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/NewRide.cpp:980]: I realise this might be a work in progress, but this needs to be moved to the language files. Rather than the `(s)` bit, I think we should just go with two strings, single and plural.

Comment (fredriktegnell) [reply src/openrct2-ui/windows/NewRide.cpp:980]: Yeah that makes sense. Just to make sure I understand you completely, this line:
`authorsString.append("Author(s): ");`
Should be divided into two separate strings in the language files, like Author and Authors? And then decide which one to use depending on single or plural.

Comment (tupaschoal) [reply src/openrct2-ui/windows/NewRide.cpp:980]: Yes, but you can worry about it later, once we converge on what the solution will be.

Comment (karst) [reply src/openrct2-ui/windows/NewRide.cpp:980]: Perhaps the solution used in #21058?

**Review by janisozaur — APPROVED** [review]

File: src/openrct2-ui/windows/NewRide.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/NewRide.cpp:980]: @fredriktegnell can you take on this issue? We agreed on proceeding with this PR

Comment (fredriktegnell) [reply src/openrct2-ui/windows/NewRide.cpp:980]: @tupaschoal Great, I'll try to get it done as soon as I've got the time.

**Comment by AaronVanGeffen** [comment]
Rebased the PR and added localisable author strings. The string is now also hidden unless debugging tools are active.

Forest Frontiers 2024-07-08 19-24-03

Other than a changelog entry, this is good to go as far as I'm concerned. I'd like someone else to test some edge cases, though.

**Review by karst — APPROVED** [review]
This all seems to work as intended!
I will copy that last line and make it work with the scenery manager and make a PR for it.
